src/assign.c

```diff
@@ -845,6 +851,8 @@ const char *memrecycle(const SEXP target, const SEXP where, const int start, con
     warning(_("Coercing 'list' RHS to '%s' to match the type of %s."), type2char(TYPEOF(target)), targetDesc);
     source = PROTECT(coerceVector(source, TYPEOF(target))); protecti++;
   } else if ((TYPEOF(target)!=TYPEOF(source) || targetIsI64!=sourceIsI64) && !isNewList(target)) {
+    static char targetDesc[501];
+    snprintf(targetDesc, 500, colnum==0 ? _("target vector") : _("column %d named '%s'"), colnum, colname);
```

Comment:
shouldn't this one be declared inside the `GetVerbose()>=3` branch?

Reply:
I think macro below needs it as well

Reply:
but the macro only needs it inside its `if(COND)`. So isn't this `snprintf()` indeed wasteful?  This `snprintf` could be inside this `GetVerbose()>=3` and then appear inside the CHECK_RANGE macro as well inside its `if(COND)`. When I didn't see the `snprintf` inside the `CHECK_RANGE` that's why I wrote https://github.com/Rdatatable/data.table/pull/5463#pullrequestreview-1140163707. So it would read better if `targetDesc` was populated right there in `CHECK_RANGE` next to where it is used?